code-experiments/src/c_linear.c

```diff
@@ -310,22 +317,31 @@ static coco_problem_t *c_linear_single_cons_bbob_problem_allocate(const size_t f
                                 + 50000 * constraint_number);
   srand(seed_cons_i);
   
+  /* The constraints gradients are scaled with random numbers
+   * 10**U[0,1] and 10**U_i[0,2], where U[a, b] is uniform in [a,b] 
+   * and only U_i is drawn for each constraint individually. 
+   * The random number 10**U[0,1] is given by the variable 'factor1' 
+   * while the random number 10**U_i[0,2] is calculated below and 
+   * stored as 'factor2'. (The exception is when the number of
+   * constraints is n+1, in which case 'factor2' defines a random
+   * number 10**U_i[0,1])
+   */
+     
+  if (number_of_linear_constraints == dimension + 1)
+    exp2 = rand() / (RAND_MAX / (1 + 1) + 1);
```

Comment:
Do we have an issue for this bug? 

Unfortunately any change of the "random" number generator can only happen before the code goes into the release, because the instances should never change again afterwards. 

On a similar note, we probably have no guaranty that the library function produces always and on all platforms exactly the same numbers. Its contract seems rather to be that it produces always a sequence of numbers with apparent uniform distribution. I believe this is one of the main reasons coco has its own generator in the first place.